pkg/controller/container-runtime-config/helpers.go

```diff
@@ -511,15 +513,45 @@ func getValidBlockedRegistries(releaseImage string, imgSpec *apicfgv1.ImageSpec)
 	for i, reg := range imgSpec.RegistrySources.BlockedRegistries {
 		// if there is a match, return all the blocked registries except the one that matched and return an error as well
 		if reg == payloadReg {
-			blockedRegs = append(blockedRegs, imgSpec.RegistrySources.BlockedRegistries[i+1:]...)
-			return blockedRegs, fmt.Errorf("error adding %q to blocked registries, cannot block the registry being used by the payload", payloadReg)
+			// If the payload registry doesn't have mirror rules configured for it, then don't add it to the blocked registries list
+			// TODO maybe? Add an opt-in configuration to ICSP and check if that is set as well
+			if !payloadRegHasMirrorConfig(payloadReg, icspRules, imgSpec.RegistrySources.BlockedRegistries) {
+				blockedRegs = append(blockedRegs, imgSpec.RegistrySources.BlockedRegistries[i+1:]...)
+				return blockedRegs, fmt.Errorf("error adding %q to blocked registries, cannot block the registry being used by the payload", payloadReg)
+			}
+			// Log a warning that we are adding the payload registries to the blocked registries list as there are mirror rules for it
+			glog.Warningf("%q is the payload registry, but will add it to the list of blocked registries as there are mirror rules configured for it", reg)
```

Comment:
Is this a warning-level event? Everything is working as intended. (OTOH I haven’t taken the time to look up any guidelines about log levels.)

Reply:
This warning would show up in the machine-config-controller logs. I would like to log a warning when we block the payload registry so that in the event where the cluster goes down because something is wrong with the mirrors configured etc, the logs can show that the payload registry was blocked and maybe that is why things are breaking.

Reply:
That’s a good point.